gstaichi/program/launch_context_builder.cpp

```diff
@@ -242,6 +244,33 @@ void LaunchContextBuilder::set_arg_ndarray(const std::vector<int> &arg_id,
   set_arg_ndarray_impl(arg_id, ptr, arr.shape);
 }
 
+void LaunchContextBuilder::set_args_ndarray(
+    const std::vector<int> &arg_ids,
+    const std::vector<Ndarray *> &arrs) {
+  const size_t num_arrs = arg_ids.size();
+
+  array_ptrs.reserve(array_ptrs.size() + num_arrs);
+  array_runtime_sizes.reserve(array_runtime_sizes.size() + num_arrs);
+  device_allocation_type.reserve(device_allocation_type.size() + num_arrs);
+
+  TI_ASSERT(num_arrs == arg_ids.size());
+  for (int i = 0; i < num_arrs; ++i) {
+    const int arg_id = arg_ids[i];
+    const Ndarray &arr = *arrs[i];
+
+    intptr_t ptr = arr.get_device_allocation_ptr_as_int();
+    array_ptrs[{arg_id, TypeFactory::DATA_PTR_POS_IN_NDARRAY}] = (void *)ptr;
+    set_array_device_allocation_type({arg_id}, DevAllocType::kNdarray);
+    const std::vector<int> &shape = arr.shape;
+    size_t total_size = 1;
```

Comment:
I wonder if converting the shape into totla size should be factorized out into a separate function?

Reply:
oh, there's also the set_struct_arg function being called though 🤔

Reply:
As for the rest, just a copy paste at this point.